photosynthesis_metrics/gs.py

```diff
@@ -0,0 +1,255 @@
+r""" This module implements Geometry Score (GS) in PyTorch.
+
+Implementation is inspired by Valentin Khrulkov's (@KhrulkovV) implementation:
+https://github.com/KhrulkovV/geometry-score
+
+See paper for details:
+https://arxiv.org/pdf/1802.02664.pdf
+"""
+from typing import Optional
+from functools import partial
+from multiprocessing import Pool
+
+import torch
+import numpy as np
+from scipy.spatial.distance import cdist
+
+from photosynthesis_metrics.base import BaseFeatureMetric
+
+
+def relative(intervals: np.ndarray, alpha_max: float, i_max: int = 100) -> np.ndarray:
+    r"""
+    For a collection of intervals this functions computes
+    RLT by formulas (2) and (3) from the paper. This function will be typically called
+    on the output of the gudhi persistence_intervals_in_dimension function.
+
+    Args:
+      intervals: list of intervals e.g. [[0, 1], [0, 2], [0, np.inf]].
+      alpha_max: The maximal persistence value
+      i_max: Upper bound on the value of beta_1 to compute.
+
+    Returns:
+        rlt: Array of size (i_max, ) containing desired RLT.
+    """
+
+    persistence_intervals = []
+    # If for some interval we have that it persisted up to np.inf
+    # we replace this point with alpha_max.
+    for interval in intervals:
+        if not np.isinf(interval[1]):
+            persistence_intervals.append(list(interval))
+        elif np.isinf(interval[1]):
+            persistence_intervals.append([interval[0], alpha_max])
+
+    # If there are no intervals in H1 then we always observed 0 holes.
+    if len(persistence_intervals) == 0:
+        rlt = np.zeros(i_max)
+        rlt[0] = 1.0
+        return rlt
```

Comment:
`rlt` matches function's name, which can be confusing. Try to use a different name if possible.
Also `rlt` is a bad name in general because it is hard to understand what stands behind it. Try to change this name if possible